### PR TITLE
Allow specifying a package override for pydocgen

### PR DESCRIPTION
**Pull request by praneetloke**
### Proposed changes

As the title states, this PR contains changes to the `generate_python_docs.sh` script to allow passing an optional argument to override the package name. I have also updated `pydocgen` to accept the same optional argument as well and build docs for just that package.

I have also added some missing package names to the following scripts so that they can be run to generate all docs. I kind of went down the rabbit hole with this one:
* `run_typedoc.sh`
* `generate_python_docs.sh`
* `generate_all_resource_docs.sh`
* `docfx/docfx.json`

Along with that I also updated our README with instructions on how to enlist a new provider for docs since that information seemed to be missing.

EDIT: Oh and once this is merged, we'll need to update https://github.com/pulumi/scripts/blob/master/ci/build-package-docs.sh#L77 to pass the provider name.

### Unreleased product version (optional)

N/A

### Related issues (optional)

N/A

### CODE REVIEW

File: docfx/docfx.json [hunk 1]
```diff
@@ -208,6 +226,15 @@
...
       "dest": "metadata/Pulumi.Mailgun"
     },
+    {
+      "src": [
+        {
+          "src": "../../pulumi-mongodbatlas/sdk/dotnet",
+          "files": ["Pulumi.*.csproj"]
+        }
+      ],
+      "dest": "metadata/Pulumi.Mongodbatlas"
```
Comment (praneetloke) [inline docfx/docfx.json:236]: I got this name by looking at our .NET SDK for this provider and checked the namespace for one of the C# classes.

File: docfx/docfx.json [hunk 2]
```diff
@@ -163,6 +163,15 @@
...
       "dest": "metadata/Pulumi.F5BigIP"
     },
+    {
+      "src": [
+        {
+          "src": "../../pulumi-github/sdk/dotnet",
+          "files": ["Pulumi.*.csproj"]
+        }
+      ],
+      "dest": "metadata/Pulumi.Github"
```
Comment (praneetloke) [inline docfx/docfx.json:173]: I know this way of spelling GitHub is not the right way, but alas, that's the name we used in our .NET SDK per one of the C# classes that I looked at.

File: README.md [hunk 3]
```diff
@@ -157,6 +157,26 @@ We currently have CI in-place to call the above script whenever a new provider v
...
+Since Pulumi uses CI to generate docs, there isn't a need to keep scripts up-to-date when docs for a new provider
+ need to be generated. Every provider repo uses similar `Makefile`  that contains calls to the
+  relevant scripts to generate docs for that provider when a new version is released.
+
+However, there may be times when you need to generate docs for all providers out-of-band and submit a PR for it. To
+ do that, ensure that the following files have all of the providers listed:
+* For NodeJS API docs: `scripts/run_typedoc.sh`
+* For Python API docs: `scripts/generate_python_docs.sh` and `tools/pydocgen/pulumi-docs.json`
+* For .NET API docs: `docfx/docfx.json`
+* For Resource docs: `scripts/generate_all_resource_docs.sh`
```
Comment (justinvp) [inline README.md:172]: Lower priority, but we really should consolidate these into a single place. Should we open an issue?

File: tools/pydocgen/pulumi-docs.json [hunk 4]
```diff
@@ -115,6 +115,12 @@
       "pulumi_provider_name": "keycloak",
       "terraform_provider_name": ""
     },
+    {
+      "name": "Pulumi Kong",
+      "package_name": "pulumi_kong",
+      "pulumi_provider_name": "kong",
+      "terraform_provider_name": "kong"
```
Comment (justinvp) [inline tools/pydocgen/pulumi-docs.json:122]: We leave this blank when the TF provider isn't in the https://github.com/terraform-providers GH org.

```suggestion
      "terraform_provider_name": ""
```

File: README.md (hunk 3 above)
Comment (praneetloke) [reply README.md:172]: Yes, definitely! I was thinking the exact thing while I was typing this out. But yeah this is the current situation. And yes, I'll open an issue for it.

File: tools/pydocgen/pulumi-docs.json (hunk 4 above)
Comment (praneetloke) [reply tools/pydocgen/pulumi-docs.json:122]: Ah ok. I'll update it.

File: README.md (hunk 3 above)
Comment (praneetloke) [reply README.md:172]: Created https://github.com/pulumi/docs/issues/3168 for this.

**Pull request merged**